test/powershell/Modules/Microsoft.PowerShell.Management/Clear-Content.Tests.ps1

```diff
@@ -1,26 +1,171 @@
+    $functionName
+}
+
 Describe "Clear-Content cmdlet tests" -Tags "CI" {
-  $file1 = "file1.txt"
-  Setup -File "$file1"
+  BeforeAll {
+    $file1 = "file1.txt"
+    $file2 = "file2.txt"
+    $file3 = "file3.txt"
+    $content1 = "This is content"
+    $content2 = "This is content for alternate stream tests"
+    Setup -File "$file1"
+    Setup -File "$file2" -Content $content1
+    Setup -File "$file3" -Content $content2
+    $streamContent = "content for alternate stream"
+    $streamName = "altStream1"
+  }
 
   Context "Clear-Content should actually clear content" {
     It "should clear-Content of testdrive:\$file1" {
       set-content -path testdrive:\$file1 -value "ExpectedContent" -passthru | Should be "ExpectedContent"
       clear-content -Path testdrive:\$file1
     }
+
     It "shouldn't get any content from testdrive:\$file1" {
       $result = get-content -path testdrive:\$file1
       $result | Should BeExactly $null
     }
+
+    # we could suppress the WhatIf output here if we use the testhost, but it's not necessary
+    It "The filesystem provider supports should process" -skip:(!$IsWindows) {
+      clear-content TESTDRIVE:\$file2 -WhatIf
+      "TESTDRIVE:\$file2" | should contain "This is content"
+    }
+
+    It "The filesystem provider should support ShouldProcess (reference ProviderSupportsShouldProcess member)" {
+      $cci = ((get-command clear-content).ImplementingType)::new()
+      $cci.SupportsShouldProcess | should be $true
+    }
+
+    It "Alternate streams should be cleared with clear-content" -skip:(!$IsWindows) {
+      # make sure that the content is correct
+      # this is here rather than BeforeAll because only windows can write to an alternate stream
+      set-content -path "TESTDRIVE:/$file3" -stream $streamName -value $streamContent
+      get-content -path "TESTDRIVE:/$file3" | Should be $content2
+      get-content -Path "TESTDRIVE:/$file3" -stream $streamName | should be $streamContent
+      clear-content -PATH "TESTDRIVE:/$file3" -stream $streamName
+      get-content -Path "TESTDRIVE:/$file3" | should be $content2
+      get-content -Path "TESTDRIVE:/$file3" -stream $streamName | should BeNullOrEmpty
+    }
+
+    It "the '-Stream' dynamic parameter is visible to get-command in the filesystem" {
+      try {
+        push-location TESTDRIVE:
+        (get-command clear-content -stream foo).parameters.keys -eq "stream" | should be "stream"
+      }
+      finally {
+        pop-location
+      }
+    }
+
+    It "the '-stream' dynamic parameter should not be visible to get-command in the function provider" {
+      try {
+        push-location function:
+        get-command clear-content -stream $streamName
+        throw "ExpectedExceptionNotDelivered"
+      }
+      catch {
+        $_.FullyQualifiedErrorId | should be "NamedParameterNotFound,Microsoft.PowerShell.Commands.GetCommandCommand"
+      }
+      finally {
+        pop-location
+      }
```

Comment:
I have a bit of a problem with this. The functionality of checking the FullyQualifiedErrorId rather than the exception message should be in Pester rather than our environment.  Putting this code in our own environment does not help the community, but only increases the burden on our code maintenance.